### PR TITLE
Fixed visual update bug where hearts weren't displaying correctly after taking damage

### PR DESCRIPTION
**Pull request by dev-pengu**
taking damage

### CODE REVIEW

File: source/actors/player/PlayerHealth.hx [hunk 1]
```diff
@@ -37,14 +37,17 @@ class PlayerHealth extends FlxTypedGroup<HeartContainer> {
     }
 
     public function updateGraphics(currentHealth:Float):Void {
-        for (val in 0...countLiving()) {
+        for (val in 0...length) {
+            trace("Health at call " + (val + 1) + ": " + currentHealth);
```
Comment (SamBumgardner) [inline source/actors/player/PlayerHealth.hx]: Let's strip out this logging line before merging
```suggestion
```

File: source/actors/player/PlayerHealth.hx [hunk 2]
```diff
@@ -37,14 +37,17 @@ class PlayerHealth extends FlxTypedGroup<HeartContainer> {
...
             var heart:HeartContainer = members[val];
             if (currentHealth >= HeartContainer.MAX_FRAG_AMOUNT) {
                 currentHealth -= HeartContainer.MAX_FRAG_AMOUNT;
                 heart.setFragments(HeartContainer.MAX_FRAG_AMOUNT);
-            } else {
+            } else if (currentHealth > 0) {
                 heart.setFragments(Std.int(currentHealth));
-                break;
+                currentHealth -= currentHealth;
```
Comment (SamBumgardner) [inline source/actors/player/PlayerHealth.hx]: After re-reading over the code a couple of times, this makes sense (the only way we reach this conditional is if we're filling in the last partially filled heart) but it read very strangely to me during my first pass.

There's no need to change anything, but I do wish there was a way to make this a little easier to read. 

Maybe it'd be more straightforward (albeit involve a bit more calculation) if currentHealth didn't change during loop iterations? I think most of my confusion came from that. Maybe each cycle through the loop could just do something like

```
for (i in 0...length) {
    var healthToDisplay = currentHealth - i * HeartContainer.MAX_FRAG_AMOUNT 
    members[i].setFragments(Math.max(healthToDisplay, 0));
}
```

File: source/actors/player/PlayerHealth.hx (hunk 1 above)
Comment (dev-pengu) [reply source/actors/player/PlayerHealth.hx]: I got that line removed

File: source/actors/player/PlayerHealth.hx (hunk 2 above)
Comment (dev-pengu) [reply source/actors/player/PlayerHealth.hx]: That makes sense. I forgot I had a safeguard in the HeartContainer to take care of excess fragment value. I tested that out and it worked well and I agree it is more readable.

**Review by SamBumgardner — APPROVED** [review]

**Pull request merged**